glean-core/tests/common/mod.rs

```diff
@@ -0,0 +1,29 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use glean_core::Glean;
+
+// #[allow(dead_code)] is required on functions in this module as a workaround
+// for https://github.com/rust-lang/rust/issues/46379
```

Comment:
@badboy: The issue sort of makes sense given that test modules are compiled as separate crates. I still think the upside of putting this kind of code in a common place outweighs the downsides of having declare everything with #[allow(dead_code)], but if you have any alternative suggestions, let me know.